detectors/gcp/detector.go

```diff
@@ -144,6 +145,31 @@ func (r *resourceBuilder) addZoneOrRegion(detect func() (string, gcp.LocationTyp
 	}
 }
 
+var (
+	// TODO: semconv.GCPGceInstanceGroupManagerNameKey
+	gcpGceInstanceGroupManagerNameKey = attribute.Key("gcp.gce.instance_group_manager.name")
+	// TODO: semconv.GCPGceInstanceGroupManagerZoneKey
+	gcpGceInstanceGroupManagerZoneKey = attribute.Key("gcp.gce.instance_group_manager.zone")
+	// TODO: semconv.GCPGceInstanceGroupManagerRegionKey
+	gcpGceInstanceGroupManagerRegionKey = attribute.Key("gcp.gce.instance_group_manager.region")
+)
+
+func (r *resourceBuilder) addMIG(detect func() (gcp.ManagedInstanceGroup, error)) {
+	if mig, err := detect(); err == nil {
+		if mig.Name != "" {
+			r.attrs = append(r.attrs, gcpGceInstanceGroupManagerNameKey.String(mig.Name))
+		}
+		switch mig.Type {
+		case gcp.Zone:
+			r.attrs = append(r.attrs, gcpGceInstanceGroupManagerZoneKey.String(mig.Location))
+		case gcp.Region:
+			r.attrs = append(r.attrs, gcpGceInstanceGroupManagerRegionKey.String(mig.Location))
+		}
+	} else {
+		r.errs = append(r.errs, err)
+	}
```

Comment:
Indentation can be reduced by restructuring:

```suggestion
	mig, err := detect()
	if err != nil {
		r.errs = append(r.errs, err)
		return
	}
	if mig.Name != "" {
		r.attrs = append(r.attrs, gcpGceInstanceGroupManagerNameKey.String(mig.Name))
	}
	switch mig.Type {
	case gcp.Zone:
		r.attrs = append(r.attrs, gcpGceInstanceGroupManagerZoneKey.String(mig.Location))
	case gcp.Region:
		r.attrs = append(r.attrs, gcpGceInstanceGroupManagerRegionKey.String(mig.Location))
	}
```